### PR TITLE
Make ride construction shortcuts work for footpath too

### PR DESCRIPTION
**Pull request by telk5093**
This allows the ride construction shortcuts work for footpath, too.
I'm not good at understanding OpenRCT2's code so please let me know if something is not fit with project.

~~I have considered that renaming ``ShortcutRideConstruction(~~)`` into ``ShortcutConstruction(~~)``, but footpath construction only uses part of them so I left thier name for now.~~

Please give me comments how do you feel this feature. Thxs :)

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
@IntelOrca will this get in the way of the shortcut refactor

**Comment by telk5093** [comment]
I've solved conflicts from the shortcut refactor, but I'm not sure you devs like this 😅

**Comment by IntelOrca** [comment]
The changes look alright, but do you plan on creating or changing the strings to be more generic as well as currently they still refer to the shortcuts as ride construction.

**Comment by telk5093** [comment]
Thanks, I didn't noticed there it is.
I just changed only StringIds since it didn't work for me if I change ShortcutIds' name such as ``ShortcutId::WindowRideConstructionTurnLeft``. It is out of my skill 😅

File: src/openrct2-ui/windows/Footpath.cpp [hunk 1]
```diff
@@ -1242,3 +1242,100 @@ static bool footpath_select_default()
         return true;
     }
 }
+
+void window_footpath_keyboard_shortcut_turn_left()
+{
+    rct_window* w = window_find_by_class(WC_FOOTPATH);
+    if (w == nullptr || WidgetIsDisabled(w, WIDX_DIRECTION_NW) || WidgetIsDisabled(w, WIDX_DIRECTION_NE)
+        || WidgetIsDisabled(w, WIDX_DIRECTION_SW) || WidgetIsDisabled(w, WIDX_DIRECTION_SE)
+        || gFootpathConstructionMode != 2)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Footpath.cpp:1251]:
```suggestion
        || WidgetIsDisabled(w, WIDX_DIRECTION_SW) || WidgetIsDisabled(w, WIDX_DIRECTION_SE) || gFootpathConstructionMode != 2)
```
To please the code formatting check

File: src/openrct2-ui/windows/Footpath.cpp [hunk 2]
```diff
@@ -1242,3 +1242,100 @@ static bool footpath_select_default()
...
+    int32_t turnedRotation = gFootpathConstructDirection - currentRotation + (currentRotation % 2 == 1 ? 1 : -1);
+    window_footpath_mousedown_direction(turnedRotation);
+}
+
+void window_footpath_keyboard_shortcut_turn_right()
+{
+    rct_window* w = window_find_by_class(WC_FOOTPATH);
+    if (w == nullptr || WidgetIsDisabled(w, WIDX_DIRECTION_NW) || WidgetIsDisabled(w, WIDX_DIRECTION_NE)
+        || WidgetIsDisabled(w, WIDX_DIRECTION_SW) || WidgetIsDisabled(w, WIDX_DIRECTION_SE)
+        || gFootpathConstructionMode != 2)
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Footpath.cpp:1265]:
```suggestion
        || WidgetIsDisabled(w, WIDX_DIRECTION_SW) || WidgetIsDisabled(w, WIDX_DIRECTION_SE) || gFootpathConstructionMode != 2)
```
To please the code formatting check

**Comment by IntelOrca** [comment]
@telk5093 did you intend to also change the strings? We change `en-GB.txt` in the main repo. Other that, it seems to work nicely.

**Review by IntelOrca — APPROVED** [review]

**Pull request merged**